pkg/drawio/createMapFile.go

```diff
@@ -58,52 +119,70 @@ func (data *templateData) AY(tn TreeNodeInterface) int {
 	return y
 }
 
-// orderNodesForTemplate() sort the nodes for the drawio/svg canvas
-// the order in the drawio/svg canvas are set by the order in the drawio/svg file
+// orderNodesForTemplate() sort the nodes for the drawio/svg/html canvas
+// the order in the canvas are set by the order in the drawio/svg/html file
 // (the last in the file will be on top in the canvas)
+// for clicking on a node, it must not be behind another node
 // 1. we put the lines at the top so they will overlap the icons
-// 2. we put the icons above the squares so we can mouse over it for tooltips
-// 3. we put the sgs and the gs in the bottom.
-// (if a sg ot a gs is above a square, it will block the the tooltip of the children of the square.)
+// 2. we put the icons above the squares
+// 3. we bucket sort the squares, and order them by parent-child order
+// 4. we also sort the groupSquare by size
 func orderNodesForTemplate(nodes []TreeNodeInterface) []TreeNodeInterface {
-	var sg, sq, ln, ic, gs, orderedNodes []TreeNodeInterface
+	squareOrders := []SquareTreeNodeInterface{
+		&NetworkTreeNode{},
+		&PublicNetworkTreeNode{},
+		&CloudTreeNode{},
+		&RegionTreeNode{},
+		&VpcTreeNode{},
+		&GroupSubnetsSquareTreeNode{},
+		&ZoneTreeNode{},
+		&SubnetTreeNode{},
+		&SGTreeNode{},
+		&PartialSGTreeNode{},
+		&GroupSquareTreeNode{},
+	}
+	var lines, icons, orderedNodes []TreeNodeInterface
+	squaresBuckets := map[reflect.Type][]TreeNodeInterface{}
+	for _, t := range squareOrders {
+		squaresBuckets[reflect.TypeOf(t).Elem()] = []TreeNodeInterface{}
+	}
```

Comment:
Is this initialization needed?
There is no problem in calling `append` with `nil` as the first argument